fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaData.java

```diff
@@ -372,7 +372,7 @@ public boolean usesSubspaceKeyCounter() {
 
     @Nonnull
     public Collection<RecordType> recordTypesForIndex(@Nonnull Index index) {
-        if (getUniversalIndexes().contains(index)) {
+        if (hasUniversalIndex(index.getName())) {
             return getRecordTypes().values();
         }
         List<RecordType> result = new ArrayList<>();
```

Comment:
Would it make sense to cache the result of this function back inside the `Index` itself?
For most indexes, it will be the singleton, so that's not much more space and the meta-data has a reasonably long lifetime.

Reply:
I can take a look at @MMcM

Reply:
@MMcM I can cache it back but it seems out of place (universal logic is non-existent on the Index).  Let me know if you need me to do that.

Reply:
Alternatively, instead of putting it in the index, it could go in the meta-data, constructed by the builder so that the `Index` doesn't need to have any notion of what types exist in the meta-data

Reply:
Yes, caching in the `RecordMetaData` itself work just fine. There is precedent for caching in the individual meta-data objects with things like `RecordType.recordTypeKey`.

Reply:
@MMcM and @alecgrieser is this still true for an Index?

```
   @Override
    public int hashCode() {
        // Within the context of a single RecordMetaData, this should be sufficient
        return Objects.hash(name, type);
    }
```

Reply:
That would still be true, I suppose, though in a single `RecordMetaData`, `name.hashCode()` would also be sufficient, I would think